src/main/java/com/example/java/test/junior/developer/mapper/CategoryMapper.java

```diff
@@ -2,12 +2,23 @@
 
 import com.example.java.test.junior.developer.dto.CategoryDto;
 import com.example.java.test.junior.developer.model.Category;
+import com.example.java.test.junior.developer.service.LanguageService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class CategoryMapper {
+
+  private final LanguageService languageService;
+
+  @Autowired
+  public CategoryMapper(LanguageService languageService) {
+    this.languageService = languageService;
+  }
+
   public Category toEntity(CategoryDto dto) {
     return Category.builder()
+        .language(languageService.getLanguage(dto.getId()))
```

Comment:
This is not correct. Id in CategoryDto refers to CATEGORY id, and not language id. You should add languageId field to CategoryDto and use it to find language.